lib/searchyll/configuration.rb

```diff
@@ -84,18 +84,15 @@ def elasticsearch_mapping
     end
 
     def elasticsearch_settings
-      shards = site.config['elasticsearch']['number_of_shards'] || 1
-      replicas = site.config['elasticsearch']['number_of_replicas'] || 1
-      read_yaml(elasticsearch_settings_path, {
-        'index' => {
-          'number_of_shards'   => shards,
-          'number_of_replicas' => replicas,
-          'refresh_interval'   => -1
-        }
-      })
+      settings = read_yaml(elasticsearch_settings_path)
+      settings['index'] ||= {}
+      settings['index']['number_of_shards']   ||= site.config['elasticsearch']['number_of_shards'] || 1
+      settings['index']['number_of_replicas'] ||= site.config['elasticsearch']['number_of_replicas'] || 1
```

Comment:
Good setting here. We should move default settings to a separate YAML and read it from there instead of hard-coding - will add TODO issue for that.